lib/blinkist-airbrake-scrubber/deep_traversal.rb

```diff
@@ -0,0 +1,33 @@
+# DeepTraversal provides traverse possibility of Hashes
+# Can handle Hash objects with deep nesting, or other nested objects that can be dug deeper (e.g. Array)
+class DeepTraversal
+  def initialize(source)
+    @source = source
+  end
+
+  def traverse(&block)
+    recursive_traverse(@source, &block)
+  end
+
+  private
+
+  def recursive_traverse(input, &block)
+    if input.is_a? Array
+      input.map { |i| recursive_traverse(i, &block) }
+
+    elsif input.respond_to?(:to_hash)
+      Hash[input.map { |key, value|
```

Comment:
This is wrong: not everything that responds to `to_hash` will be a hash and will implement `.map` method. A little bit less wrong, but still pretty wrong would be `input.to_hash.map`